ci/Jenkinsfile

```diff
@@ -70,16 +79,51 @@ pipeline {
   }
 
   stages {
-    stage('Download') {
+     stage('Download') {
       steps {
         script {
-          sh "mkdir '${env.WORKSPACE}/tmp/'"
-        }
-        fileOperations([
-          fileDownloadOperation(url:"${params.BUILD}", userName:"${CI_CREDS_USR}", password:"${CI_CREDS_PSW}", targetLocation:"./tmp", targetFileName:"StatusIm-Desktop.tar.gz", proxyHost:"", proxyPort:"")
-        ])
-        script {
-          sh "tar -zxvf '${env.WORKSPACE}/tmp/StatusIm-Desktop.tar.gz' -C '${env.WORKSPACE}/tmp'"
+          /* Creating directory for application build */
+          sh "mkdir -p '${env.WORKSPACE}/tmp/pkg/'"
+          def job_name = "${JOB_NAME}"
+          if(job_name.endsWith("nightly")){
+
+             /* Nightly regression */
+
+            if(params.BUILD_URL != ""){
+              echo "BUILD_URL value was ignored. Please use 'https://ci.infra.status.im/job/status-desktop/job/e2e/job/manual/' to test specific builds."
+            }
+            copyArtifacts(projectName: "status-desktop/systems/linux/x86_64/package/", filter: 'pkg/*-x86_64.tar.gz', selector: lastWithArtifacts(), target: "${env.WORKSPACE}/tmp")
+          } else if (job_name.endsWith("manual")){
+
+             /* Manual run and PRs trigger for desktop-qa-automation */
+
+            if(params.BUILD_URL == ""){
+              copyArtifacts(projectName: "status-desktop/systems/linux/x86_64/package/", filter: 'pkg/*-x86_64.tar.gz', selector: lastWithArtifacts(), target: "${env.WORKSPACE}/tmp")
```

Comment:
I really think this should be separate stages. This wall of code is hard to follow. It should be two mutually exclusive stages: `Copy` and `Download`(if we want to support that) with `when` clauses that decide which one will run.

And please stop making lines so long.

Reply:
I removed the copy-paste of methods, so we have only one downloading and one copy